google/cloud/bigtable/data/_metrics/data_model.py

```diff
@@ -0,0 +1,451 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+from typing import ClassVar, Tuple, cast, TYPE_CHECKING
+
+import time
+import re
+import logging
+import uuid
+import contextvars
+
+from enum import Enum
+from functools import lru_cache
+from dataclasses import dataclass
+from dataclasses import field
+from grpc import StatusCode
+from grpc import RpcError
+from grpc.aio import AioRpcError
+
+import google.cloud.bigtable.data.exceptions as bt_exceptions
+from google.cloud.bigtable_v2.types.response_params import ResponseParams
+from google.cloud.bigtable.data._helpers import TrackedBackoffGenerator
+from google.protobuf.message import DecodeError
+
+if TYPE_CHECKING:
+    from google.cloud.bigtable.data._metrics.handlers._base import MetricsHandler
+
+
+LOGGER = logging.getLogger(__name__)
+
+# default values for zone and cluster data, if not captured
+DEFAULT_ZONE = "global"
+DEFAULT_CLUSTER_ID = "unspecified"
+
+# keys for parsing metadata blobs
+BIGTABLE_METADATA_KEY = "x-goog-ext-425905942-bin"
```

Comment:
nit: maybe a more descriptive name like BIGTABLE_LOCATION_METADATA_KEY?

Reply:
sounds good, I'll change this